### PR TITLE
Release 2.1.0

### LINKED ISSUE #344: Release version 2.1.0
### Is your enhancement related to a problem? Please describe.

This issue is for tracking changes for the `2.1.0` release. Target release date: **June 1 2023**.

### [Release steps](https://github.com/10up/safe-redirect-manager/blob/develop/CONTRIBUTING.md#release-instructions)
- [x] Branch: Starting from `develop`, cut a release branch named `release/2.1.0` for your changes.
- [x] Version bump: Bump the version number in `safe-redirect-manager.php`, `package-lock.json`, `package.json`, and `readme.txt` if it does not already reflect the version being released.  In `safe-redirect-manager.php` update both the plugin "Version:" property and the plugin `SRM_VERSION` constant.
- [x] Changelog: Add/update the changelog in both `readme.txt` and `CHANGELOG.md`.
- [x] Props: Update `CREDITS.md` file with any new contributors, confirm maintainers are accurate.
- [x] Readme updates: Make any other readme changes as necessary.  `README.md` is geared toward GitHub and `readme.txt` contains WordPress.org-specific content.  The two are slightly different.
- [x] New files: Check to be sure any new files/paths that are unnecessary in the production version are included in `.gitattributes`.
- [x] Merge: Make a non-fast-forward merge from your release branch to `develop` (or merge the pull request), then do the same for `develop` into `trunk`, ensuring you pull the most recent changes into `develop` first (`git checkout develop && git pull origin develop && git checkout trunk && git merge --no-ff develop`).  `trunk` contains the stable development version.
- [x] Push: Push your `trunk` branch to GitHub (e.g. `git push origin trunk`).
- [x] [Compare](https://github.com/10up/safe-redirect-manager/compare/trunk...develop) `trunk` to `develop` to ensure no additional changes were missed.
- [x] Test the pre-release ZIP locally by [downloading](https://github.com/10up/safe-redirect-manager/actions/workflows/build-release-zip.yml) it from the Build release zip action artifact and installing it locally. Ensure this zip has all the files we expect, that it installs and activates correctly and that all basic functionality is working.
- [x] Release: Create a [new release](https://github.com/10up/safe-redirect-manager/releases/new), naming the tag and the release with the new version number, and targeting the `trunk` branch.  Paste the changelog from `CHANGELOG.md` into the body of the release and include a link to the closed issues on the milestone (e.g. `https://github.com/10up/safe-redirect-manager/milestone/19?closed=1`).
- [x] SVN: Wait for the [GitHub Action](https://github.com/10up/safe-redirect-manager/actions?query=workflow%3A%22Deploy+to+WordPress.org%22) to finish deploying to the WordPress.org repository.  If all goes well, users with SVN commit access for that plugin will receive an emailed diff of changes.
- [x] Check WordPress.org: Ensure that the changes are live on [https://wordpress.org/plugins/safe-redirect-manager/](https://wordpress.org/plugins/safe-redirect-manager/).  This may take a few minutes.
- [x] Close milestone: Edit the [milestone](https://github.com/10up/safe-redirect-manager/milestone/19) with release date (in the `Due date (optional)` field) and link to GitHub release (in the `Description` field), then close the milestone.
- [x] Punt incomplete items: If any open issues or PRs which were milestoned for `2.1.0` do not make it into the release, update their milestone to `2.1.1` or `Future Release`.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Sidsector9**
Closes #344 

### Credits

Props @Sidsector9 @ravinderk

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -53,6 +53,27 @@ This should be a path (i.e. `/test`) or a URL (i.e. `http://example.com/wp/test`
 
 == Changelog ==
```
Comment (peterwilsoncc) [inline readme.txt:54]: This is getting quite long, as [files over 10KB can cause problems](https://developer.wordpress.org/plugins/wordpress-org/how-your-readme-txt-works/#file-size) I recommend removing all the entries for 1.x versions and replacing them with a link to the changelog on GitHub.

The file is currently 28K.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "safe-redirect-manager",
-  "version": "2.0.1",
+  "version": "2.1.0",
```
Comment (peterwilsoncc) [inline package.json:3]: This needs to be updated in package-lock.json too, in two places:

https://github.com/10up/safe-redirect-manager/blob/620210a00745f396c3a13ef71841a38068520991/package-lock.json#L3

https://github.com/10up/safe-redirect-manager/blob/620210a00745f396c3a13ef71841a38068520991/package-lock.json#L9

**Review by peterwilsoncc — COMMENTED** [review]
Added a couple of notes inline. 

In terms of release, the main one is about updating the lock file version.

File: readme.txt [hunk 3]
```diff
@@ -82,147 +103,5 @@ This should be a path (i.e. `/test`) or a URL (i.e. `http://example.com/wp/test`
...
-*   safe-redirect-manager.php - plugin_url() used properly, is_plugin_page function
-
-= 1.0 - 2012-08-27 =
-*   Plugin released
-
-== Upgrade Notice ==
-
-= 2.0.0 =
-Note that this version bumps the PHP minimum from 5.6 to 7.4 and the WordPress minimum from 4.6 to 5.7.
+= Earlier versions =
```
Comment (jeffpaul) [inline readme.txt:108]: I'm not sure how the WP.org readme.txt parser will handle this section but let's YOLO into it and see.  If it comes out oddly formed then we can do a change just to the readme on trunk to update on WP.org

**Review by jeffpaul — APPROVED** [review]
Looks good, thanks for handling the release!

File: readme.txt (hunk 3 above)
Comment (ravinderk) [reply readme.txt:108]: @jeffpaul [WordPress SEO](https://wordpress.org/plugins/wordpress-seo/#developers) is using same. 
https://github.com/Yoast/wordpress-seo/blob/00eb29a34edd4db41288eecc63a588fb9f1a46cb/readme.txt#L379

**Pull request merged**